Added lookuptable testcases (#169)
dimagi/dimagi-qa: Lookuptable/README.md

## Commcare lookuptable Test Script

> Smoke tests ensure that the important features are working and there are no showstoppers in the build deployed to environments.\

Author: Done. Thank you !!

> Lookup table tests ensure that the important features are working and there are no showstoppers in the build deployed to environments.\

Reviewer: Let's update the description here to:

`These tests ensure that the lookup table features work as expected and that there are no regressions.`

Reviewer: Let's update the description here to:

```suggestion
These tests ensure that the [lookup table ](https://confluence.dimagi.com/display/commcarepublic/Lookup+Tables)features work as expected and that there are no regressions.

```

> The automated tests comprises of [these Lookuptable functionality.](https://docs.google.com/spreadsheets/d/1-Fqax67kLz8dIJE5UcM1iGeWCV9mAm5OH6lZwb290cQ/edit#gid=0)

Reviewer: Can we please link the master lookup table test plan here instead: https://docs.google.com/spreadsheets/d/1aTBv4zGeVuNTdnCyuleZbbpqOhPEGl0f78rcKAGWQHQ/edit#gid=1360466241?

Author: changes made . Here is the commit id for reference: 99b00a9f008a562cec8a5a54baac05797b07c2c2

## Executing Scripts

### <ins> On Local Machine </ins>

#### Setting up test environment

```sh

# create and activate a virtualenv using your preferred method. Example:
python -m venv venv
source venv/bin/activate

# install requirements
pip install -r requires.txt

```

[More on setting up virtual environments](https://confluence.dimagi.com/display/GTD/QA+and+Python+Virtual+Environments)

#### Running Tests

 -   Copy `settings-sample.cfg` to `settings.cfg` and populate `settings.cfg` for
the environment you want to test.
- Run tests using pytest command like:

```sh

# To execute all the test cases 
pytest -v --rootdir= Lookuptable/testCases

```
- You could also pass the following arguments
  - ` -n 3 --dist=loadfile` - This will run the tests parallelly in 3 instances. The number of reruns is configurable.
  - ` --reruns 1` - This will re-run the tests once in case of failures.The number of reruns is configurable too.

### <ins> Trigger Manually on Gitaction </ins>

<img align="right" width="400" src="https://user-images.githubusercontent.com/67914792/168757107-3ce9bb6a-57b5-4c15-b20d-e7883bf9ed65.PNG" alt="clone this repository" />

To manually trigger the script,

>   - Go to [HQ Smoke Tests action](https://github.com/dimagi/dimagi-qa/actions/workflows/python-app.yml)

Author: Changes made and pushed the new changes . Thank you !!

>   - Go to [Lookuptable action](https://github.com/sameena1415/dimagi-qa/actions/lookuptable.yml)

Author: Changes made and pushed the new changes. Thank you !!
